### DISCUSSION TITLE
Graph for the curent 15 minutes period

### DISCUSSION THREAD

**Discussion by Krispie78** [dbuezas/lovelace-plotly-graph-card#352] [Q&A] [answered]
Hi Guys, I'm looking for a solution to make a graph for the exact curent 15 minutes period (not the same as the previous 15 minutes).  Fe. if it's now 13h12, then the period to graph would be from 13h to 13h15. Any suggestions would be apreciated !

**Comment by dbuezas** [comment, marked as answer]
You can achieve this via `$ex` functions, see this comment https://github.com/dbuezas/lovelace-plotly-graph-card/discussions/220#discussioncomment-4832891
This comment shows how to do 24 hours, but with a bit of math you can adapt it to 15 minutes.

Let me know how it goes!

**Comment by Krispie78** [comment]
Hi, thanks for you response. I had came across '#220 (comment)', but I couldn't get it working. No you directed me there again, I gave it another try. Turns out this does the trick: time_offset: $fn () => (900000-(Date.now() % 900000)).toFixed(0) + 'ms'
Reason it didn't work at first seems to be that I added the offset somewhat at the end of the code. It does work when I put before the definition of the entities that need to be shown. Strange as the offset did work being at the end of the code when it was just a fixed number (30s). Thanks for the help !

**Reply by dbuezas** [reply]
Oh I see. The error message may not be clear enough, but it tells you what needs to be before what.

The problem arises when a function depends on another one which is defined below and has therefore not yet executed.
Sometimes this is a surprise, because a lot of default behaviors are implemented internally as $fn or $ex. In your case, the data is fetched when the entities in the yaml are parsed, and to do that the offset needs to be known beforehand 

Happy to hear you solved it!

**Reply by dbuezas** [reply]
It works when you use constant values (30s in your case) because that can be used without executing a function.
The strict ordering of functions is imposed because they can use `vars`, so running them in any other order would result in weird problems, like a var being empty even if set above
